functions/Set-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,302 @@
+
+    param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer")]
+        [object[]]$SqlInstance,
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$JobName,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$StepName,
+        [Parameter(Mandatory = $false)]
+        [string]$NewName,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('ActiveScripting', 'AnalysisCommand', 'AnalysisQuery', 'CmdExec', 'Distribution', 'LogReader', 'Merge', 'PowerShell', 'QueueReader', 'Snapshot', 'Ssis', 'TransactSql')]
+        [string]$Subsystem,
+        [Parameter(Mandatory = $false)]
+        [string]$Command,
+        [Parameter(Mandatory = $false)]
+        [int]$CmdExecSuccessCode,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('QuitWithSuccess', 'QuitWithFailure', 'GoToNextStep', 'GoToStep')]
+        [string]$OnSuccessAction,
+        [Parameter(Mandatory = $false)]
+        [int]$OnSuccessStepId,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('QuitWithSuccess', 'QuitWithFailure', 'GoToNextStep', 'GoToStep')]
+        [string]$OnFailAction,
+        [Parameter(Mandatory = $false)]
+        [int]$OnFailStepId,
+        [Parameter(Mandatory = $false)]
+        [string]$DatabaseName,
+        [Parameter(Mandatory = $false)]
+        [string]$DatabaseUserName,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryAttempts,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryInterval,
+        [Parameter(Mandatory = $false)]
+        [string]$OutputFileName,
+        [Parameter(Mandatory = $false)]
+        [string]$ProxyName,
+        [switch]$Silent
+    )
+
+    begin{
+        # Check the parameter on success step id
+        if (($OnSuccessAction -ne 'GoToStep') -and ($OnSuccessStepId -ge 1)) {
+            Stop-Function -Message "Parameter OnSuccessStepId can only be used with OnSuccessAction 'GoToStep'."  -Target $instance 
+            return
+        }
+
+        # Check the parameter on success step id
+        if (($OnFailAction -ne 'GoToStep') -and ($OnFailStepId -ge 1)) {
+            Stop-Function -Message "Parameter OnFailStepId can only be used with OnFailAction 'GoToStep'."  -Target $instance 
+            return
+        }
+    }
+
+    process {
+        foreach ($instance in $sqlinstance) {
+            # Try connecting to the instance
+            Write-Message -Message "Attempting to connect to Sql Server.." -Level Output 
+            try {
+                $Server = Connect-SqlServer -SqlServer $instance -SqlCredential $SqlCredential
+            }
+            catch {
+                Stop-Function -Message "Could not connect to Sql Server instance" -Target $instance 
+                return
```

Comment:
Instead of the interrupt version, here you want to use the continue implementation of stop function, as only processing of this item has failed. Also include the exception in it and include the server name in the message:
` Stop-Function -Message "Could not connect to Sql Server instance $instance: $($_.Exception.Message)" -Target $instance -InnerErrorRecord $_ -Continue`